### DISCUSSION TITLE
duplicate value in custom module ( like email, vat id)

### DISCUSSION THREAD

**Discussion by saba1996** [YetiForceCompany/YetiForceCRM#16804] [How to]
Hello, in which directory should the custom code be written to work?
I put it in the custom module directory, but it doesn't work. I also tried writing it in the "custom" directory.

version. 6.1.0

Also, I couldn't find where the layout files for the quick create menu window are stored.

**Comment by rskrzypczak** [comment]
Check this: https://github.com/YetiForceCompany/YetiForceCRM/blob/f62d703908f3b8cd74caae0151f0e23c2e63710f/config/Performance.php#L86-L90

**Reply by saba1996** [reply]
> Check this:
> 
> https://github.com/YetiForceCompany/YetiForceCRM/blob/f62d703908f3b8cd74caae0151f0e23c2e63710f/config/Performance.php#L86-L90

 I also activated this but not  working

**Reply by rskrzypczak** [reply]
Check: https://github.com/YetiForceCompany/YetiForceCRM/discussions/15541#discussion-3738108

**Reply by saba1996** [reply]
```
> <?php
> /**
>  * Duplicate vat id handler.
>  *
>  * @package Handler
>  *
>  * @copyright YetiForce Sp. z o.o
>  * @license YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
>  * @author Mariusz Krzaczkowski <m.krzaczkowski@yetiforce.com>
>  */
> /**
>  * Accounts_DuplicateVatId_Handler class.
>  */
> class CarsInfo_DuplicateVininfo_Handler
> {
> 	/**
> 	 * EditViewPreSave handler function.
> 	 *
> 	 * @param App\EventHandler $eventHandler
> 	 */
> 	public function editViewPreSave(App\EventHandler $eventHandler)
> 	{
> 		$recordModel = $eventHandler->getRecordModel();
> 		$response = ['result' => true];
> 		$fieldModel = $recordModel->getModule()->getFieldByName('vininfo');
> 		if ($fieldModel->isViewable() && ($vat = $recordModel->get('vininfo'))) {
> 			$queryGenerator = new \App\QueryGenerator($recordModel->getModuleName());
> 			$queryGenerator->setStateCondition('All');
> 			$queryGenerator->setFields(['id'])->permissions = false;
> 			$queryGenerator->addCondition($fieldModel->getName(), $vininfo, 'e');
> 			if ($recordModel->getId()) {
> 				$queryGenerator->addCondition('id', $recordModel->getId(), 'n');
> 			}
> 			if ($queryGenerator->createQuery()->exists()) {
> 				$response = [
> 					'result' => false,
> 					'hoverField' => 'vininfo',
> 					'message' => App\Language::translate('LBL_DUPLICATE_VININFO', $recordModel->getModuleName())
> 				];
> 			}
> 		}
> 		return $response;
> 	}
> }
```

this is my custom handler, i add vitger_eventhandler but has error, when i save record.  "please contact your administrator"

**Reply by rskrzypczak** [reply]
I don't see any errors here.

The probable cause lies:

- Incorrect file location
"modules/CarsInfo/handlers/DuplicateVatId.php"
"custom/modules/CarsInfo/handlers/DuplicateVatId.php"

- Incorrectly added entry to the database

**Reply by saba1996** [reply]
> I don't see any errors here.
> 
> The probable cause lies:
> 
> * Incorrect file location
>   "modules/CarsInfo/handlers/DuplicateVatId.php"
>   "custom/modules/CarsInfo/handlers/DuplicateVatId.php"
> * Incorrectly added entry to the database

When I do duplicate restriction in existing module it works. But when I do it in the manually created module, it gives me an error. Maybe the fault is that the manually created module in the base is stored somewhere else, not in viger.

**Reply by rskrzypczak** [reply]
in the owner_id column you should have the ID of the module in which the handler file is located.
See other handlers as described in the database and where they are located.

**Reply by saba1996** [reply]
> in the owner_id column you should have the ID of the module in which the handler file is located. See other handlers as described in the database and where they are located.

@rskrzypczak 

i did everything, but there is a error message

![image](https://user-images.githubusercontent.com/56167014/215765390-006b61d8-4ccc-4e8d-bed8-4293d9e78afb.png)
